pkg/network/configurator_installer.go

```diff
@@ -0,0 +1,57 @@
+package network
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+)
+
+const nmcDownloadURL = "https://github.com/suse-edge/nm-configurator/releases/download/v0.2.0/nmc-linux-%s"
+
+type ConfiguratorInstaller struct{}
+
+func (ConfiguratorInstaller) InstallConfigurator(imageName, installPath string) error {
+	var arch string
+
+	switch {
+	case strings.Contains(imageName, "x86_64"):
+		arch = "x86_64"
+	case strings.Contains(imageName, "aarch64"):
+		arch = "aarch64"
+	default:
+		return fmt.Errorf("failed to determine arch of image %s", imageName)
+	}
+
+	file, err := os.Create(installPath)
+	if err != nil {
+		return fmt.Errorf("creating file: %w", err)
+	}
+	defer file.Close()
+
+	if err = file.Chmod(fileio.ExecutablePerms); err != nil {
+		return fmt.Errorf("adjusting permissions: %w", err)
+	}
+
+	downloadURL := fmt.Sprintf(nmcDownloadURL, arch)
+
+	resp, err := http.Get(downloadURL) // nolint: gosec
```

Comment:
Usually `http.Get()` is not advised (as per the need to ignore the linter here). One would normally use [http.NewRequest](https://pkg.go.dev/net/http#NewRequest) and then pass that to an http client.

I was not convinced that this is the approach we'd be going for anyway so I left it at `http.Get()`. 

Nevertheless, as I've mentioned yesterday, a better idea is to just pre-install both binaries into the container image and use the appropriate one for the target OS image instead of downloading it every time. I pushed the implementation changes in order to reflect this idea.